pkg/kubectl-argo-rollouts/cmd/promote/promote.go

```diff
@@ -150,13 +149,24 @@ func getPatches(rollout *v1alpha1.Rollout, skipCurrentStep, skipAllStep, full bo
 			statusPatch = []byte(promoteFullPatch)
 		}
 	default:
-		if rollout.Spec.Paused {
-			specPatch = []byte(unpausePatch)
-		}
-		if len(rollout.Status.PauseConditions) > 0 {
-			statusPatch = []byte(clearPauseConditionsPatch)
+		if rollout.Spec.Strategy.BlueGreen != nil || rollout.Spec.Paused || len(rollout.Status.PauseConditions) > 0 {
+			if rollout.Spec.Paused {
+				specPatch = []byte(unpausePatch)
+			}
+			if len(rollout.Status.PauseConditions) > 0 {
+				statusPatch = []byte(clearPauseConditionsPatch)
+			}
+			unifiedPatch = []byte(unpauseAndClearPauseConditionsPatch)
+		} else {
+			_, index := replicasetutil.GetCurrentCanaryStep(rollout)
+			// At this point, the controller knows that the rollout is a canary with steps and GetCurrentCanaryStep returns 0 if
+			// the index is not set in the rollout
+			if *index < int32(len(rollout.Spec.Strategy.Canary.Steps)) {
+				*index++
+			}
+			statusPatch = []byte(fmt.Sprintf(setCurrentStepIndex, *index))
+			unifiedPatch = statusPatch
```

Comment:
The specPatch and statusPatch to nill out pause and pause conditions should happen regardless of strategy, so I don't think the check for: `if rollout.Spec.Strategy.BlueGreen != nil` should be added. Instead, we should check if we need to increment steps after the existing code

```
...
if len(rollout.Status.PauseConditions) > 0 {
...
}

if rollout.Spec.Strategy.Canary != nil {
    replicasetutil.GetCurrentCanaryStep(rollout)
...
}
```


With the current code, it does not handle the case where 
`spec.paused: true`
and it is at a canary step because code will only do one of the patches.

Reply:
that is an OR condition , the first condition will be true for both canary and Bluegreen and else will be for canary only..to avoid confusion, I will make those changes...
one issue is if we increment step, then pause status will be redundant..